FLT/NumberField/AdeleRing.lean

```diff
@@ -18,6 +23,141 @@ end LocallyCompact
 
 section BaseChange
 
+namespace NumberField.AdeleRing
+
+variable (K L : Type*) [Field K] [Field L] [NumberField K] [NumberField L] [Algebra K L]
+
+noncomputable instance : Algebra K (NumberField.AdeleRing L) :=
+  RingHom.toAlgebra <| (algebraMap L (NumberField.AdeleRing L)).comp <| algebraMap K L
```

Comment:
```suggestion
noncomputable instance : Algebra K (NumberField.AdeleRing (𝓞 L) L) :=
  Algebra.compHom _ (algebraMap K L)
```
I think this is better because if K=L then you might get a diamond with your approach. Indeed this suggestion breaks a proof in a good way later on (it finishes early)